src/private.rs

```diff
@@ -0,0 +1,125 @@
+use std::fmt;
+
+use super::{AbiEntry, AbiFunction, AbiMetadata, AbiRoot, RootSchema, SCHEMA_VERSION};
+
+use serde::{Deserialize, Serialize};
+
+/// Core ABI information, with schema version and identity hash.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct ChunkedAbiEntry {
+    /// Semver of the ABI schema format.
+    pub schema_version: String,
+    #[serde(flatten)]
+    pub abi: AbiEntry,
+}
+
+impl ChunkedAbiEntry {
+    pub fn new(functions: Vec<AbiFunction>, root_schema: RootSchema) -> ChunkedAbiEntry {
+        Self {
+            schema_version: SCHEMA_VERSION.to_string(),
+            abi: AbiEntry {
+                functions,
+                root_schema,
+            },
+        }
+    }
+
+    pub fn combine<I: IntoIterator<Item = ChunkedAbiEntry>>(
+        entries: I,
+    ) -> Result<ChunkedAbiEntry, AbiCombineError> {
+        let mut schema_version = None;
+        let mut functions = Vec::<AbiFunction>::new();
+
+        let mut gen = schemars::gen::SchemaGenerator::default();
+        let definitions = gen.definitions_mut();
+
+        let mut unexpected_versions = std::collections::BTreeSet::new();
+
+        for entry in entries {
+            if let Some(ref schema_version) = schema_version {
+                // should probably only disallow major version mismatch
```

Comment:
In theory the version mismatch should not happen at all, since they are all pulled from the same crate. So not sure if it makes sense to bother with distinguishing major/minor versions.